### PR TITLE
Rdcc 6212

### PR DESCRIPTION
**Pull request by sahitya-desireddy**
Implement resend invite true method

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java [hunk 1]
```diff
@@ -414,5 +415,55 @@ public ResponseEntity<StaffProfileCreationResponse> updateStaffUserProfile(@Requ
...
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+    )
+    @ApiResponses({
+            @ApiResponse(
+                    code = 200,
+                    message = "Successfully updated staff user profile",
+                    response = StaffProfileCreationResponse.class,
```
Comment (vasthav-patta) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java:430]: change swagger documentation according to API

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java [hunk 2]
```diff
@@ -403,6 +404,40 @@ void should_return_update_staff_user_with_status_code_400_invalid_roles() throws
...
+        Map<String, Object> response = caseworkerReferenceDataClient.updateStaffProfile(request,ROLE_STAFF_ADMIN);
+
+        assertThat(response).isNotNull();
+        assertThat(response.get("http_status")).isEqualTo("200 OK");
+        Map responseBody = (Map) response.get("body");
+        assertThat(responseBody.get("case_worker_id")).isNotNull();
+
+    }
+
+
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java:440]: add below tests:
1.  update user details with **is_resend_invite** false and verify user details have been updated
2. update user details with **is_resend_invite** true and verify user details have been updated
3. idam id updated in staff tables when **is_resend_invite** true

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java [hunk 3]
```diff
@@ -403,6 +404,40 @@ void should_return_update_staff_user_with_status_code_400_invalid_roles() throws
...
+        userProfilePostUserWireMockForStaffProfile(HttpStatus.CREATED);
+        request.setResendInvite(true);
+
+        caseworkerReferenceDataClient.createStaffProfile(request,ROLE_STAFF_ADMIN);
+        Map<String, Object> response = caseworkerReferenceDataClient.updateStaffProfile(request,ROLE_STAFF_ADMIN);
+
+        assertThat(response).isNotNull();
+        assertThat(response.get("http_status")).isEqualTo("200 OK");
+        Map responseBody = (Map) response.get("body");
+        assertThat(responseBody.get("case_worker_id")).isNotNull();
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java:436]: assert expected case_worker_id value

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java (hunk 2 above)
Comment (sahitya-desireddy) [reply src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java:440]: ok @rajkatla-hmcts 1 and 3 rd condition i will add it now .. but 2 nd condition is not applicable as resend invite true then user details should not be updated confirmed by @kavithagade12 . both resend invite and update should not work together ... Thank you ... fyi @SalomiGeorge

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java (hunk 3 above)
Comment (sahitya-desireddy) [reply src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java]: Done this ..Thank you ...

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java (hunk 2 above)
Comment (sahitya-desireddy) [reply src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java:465]: Thanks @rajkatla-hmcts added above integration test scenarios ..kindly review ...

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java [hunk 4]
```diff
@@ -403,6 +407,62 @@ void should_return_update_staff_user_with_status_code_400_invalid_roles() throws
...
+        assertThat(resendResponse).isNotNull();
+        assertThat(resendResponse.get("http_status")).isEqualTo("200 OK");
+        Map resendResponseBody = (Map) resendResponse.get("body");
+        assertEquals(createBody.get("case_worker_id"), resendResponseBody.get("case_worker_id"));
+
+        String path = "/profile/search-by-name";
+        ResponseEntity<SearchStaffUserResponse[]> fetchstaff = caseworkerReferenceDataClient
+                .searchStaffUserByNameExchange(path, request.getFirstName(), "1", "1",
+                        ROLE_STAFF_ADMIN);
+        assertEquals(resendResponseBody.get("case_worker_id"), fetchstaff.getBody()[0].getCaseWorkerId());
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java:463]: can you add another assert to verify createResponse case_worker_id is not same as fetchstaff case_worker_id

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java (hunk 4 above)
Comment (sahitya-desireddy) [reply src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java]: done Thnak you @rajkatla-hmcts

**Review by vasthav-patta — DISMISSED** [review]

**Review by rajkatla-hmcts — APPROVED** [review]

**Pull request closed without merging**